src/main/java/net/neoforged/neoforge/fluids/potion/PotionBucketItem.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.potion;
+
+import java.util.List;
+import net.minecraft.advancements.CriteriaTriggers;
+import net.minecraft.core.component.DataComponents;
+import net.minecraft.network.chat.Component;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.stats.Stats;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.ItemUtils;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.item.UseAnim;
+import net.minecraft.world.item.alchemy.Potion;
+import net.minecraft.world.item.alchemy.PotionContents;
+import net.minecraft.world.item.alchemy.Potions;
+import net.minecraft.world.level.ItemLike;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.neoforged.neoforge.capabilities.ICapabilityProvider;
+import net.neoforged.neoforge.fluids.capability.IFluidHandlerItem;
+
+public class PotionBucketItem extends Item {
+    public PotionBucketItem(Properties properties) {
+        super(properties);
+    }
+
+    @Override
+    public String getDescriptionId(ItemStack stack) {
+        var potion = stack.getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY).potion();
+        return Potion.getName(potion, "%s.effect.".formatted(Items.POTION.getDescriptionId()));
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, TooltipContext tooltipContext, List<Component> tooltipLines, TooltipFlag tooltipFlag) {
+        var contents = stack.get(DataComponents.POTION_CONTENTS);
+
+        if (contents != null)
+            contents.addPotionTooltip(tooltipLines::add, 1F, tooltipContext.tickRate());
+    }
+
+    @Override
+    public ItemStack getDefaultInstance() {
+        return PotionContents.createItemStack(this, Potions.WATER);
+    }
+
+    @Override
+    public ItemStack finishUsingItem(ItemStack stack, Level level, LivingEntity entity) {
+        var player = entity instanceof Player plr ? plr : null;
+
+        if (player instanceof ServerPlayer sPlayer)
+            CriteriaTriggers.CONSUME_ITEM.trigger(sPlayer, stack);
+
+        if (!level.isClientSide) {
+            stack.getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY).forEachEffect(effectInstance -> {
```

Comment:
When I said "I also made it drinkable", there are a lot of details that go into it being drinkable beyond just being a clone of the vanilla item. Notably, a bucket of potion holds 4x as much as a bottle, so it does not make sense that it acts identically to a potion bottle. However, how much more effect you get is up for debate, any value is going to be somewhat controversial.

I personally choose to make the effect 2.5x as strong as vanilla potions as that number behaved nicely with things like health potions and you get the advantage of longer durations reducing when you need to drink again. But that is just what I did, someone else might have a different opinion; I am not entirely convinced it should be Neo deciding what the choice is.